modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/api.ts

```diff
@@ -39,12 +59,24 @@ function getValidFields(properties: any): Array<IField> {
 			return;
 		}
 
-		if (propertyValue.$ref) {
+		if (type === 'object' && propertyValue.$ref) {
+			if (Liferay.FeatureFlags['LPS-186871']) {
+				fields.push({
+					children: getValidFields({
+						schemaName: propertyValue.$ref.replace(/^.*\//, ''),
+						schemas,
+					}),
```

Comment:
I made parsing recursive here, so it will work with any number of nested object. However, I don't think this is currently possible to have in Objects definition. It only works on two levels, see for example FDSSort > FDSView in opening comment video. FDSView > FDSEntry relationship is a stub.